src/hooks/useQueryWithRetry.jsx

```
@@ -0,0 +1,43 @@
+// Wraps useQuery and, when a query ends up in `failed` (a thrown fetch, not an
+// HTTP error status), retries it a few times with exponential backoff before
+// surfacing the failure. While retries remain the consumer sees `isRetrying`
+// (keep showing a spinner) rather than `hasError`.
+export const useQueryWithRetry = (definition, options) => {
+  const queryResult = useQuery(definition, options)
+  const { fetchStatus, fetch } = queryResult
+  const [retryCount, setRetryCount] = useState(0)
+
+  useEffect(() => {
+    if (fetchStatus === 'loaded') {
+      setRetryCount(count => (count === 0 ? count : 0))
```

ESLint

[error] Error: Calling setState synchronously within an effect can trigger cascading renders

Effects are intended to synchronize state between React and external systems such as manually updating the DOM, state management libraries, or other platform APIs. In general, the body of an effect should do one or both of the following:
* Update external systems with the latest state from React.
* Subscribe for updates from some external system, calling setState in a callback function when external state changes.

Calling setState synchronously within an effect body causes cascading renders that can hurt performance, and is not recommended. (https://react.dev/learn/you-might-not-need-an-effect).

  23 |   useEffect(() => {
  24 |     if (fetchStatus === 'loaded') {
> 25 |       setRetryCount(count => (count === 0 ? count : 0))
     |       ^^^^^^^^^^^^^ Avoid calling setState() directly within an effect
  26 |       return
  27 |     }
  28 |     if (fetchStatus !== 'failed' || retryCount >= MAX_RETRIES) return

(react-hooks/set-state-in-effect)